src/context/FeaturebaseContext/FeaturebaseContextProvider.tsx

```diff
@@ -0,0 +1,99 @@
+import React, {
+  type FC,
+  type PropsWithChildren,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
+import { useAppContext } from '~context/AppContext/AppContext.ts';
+import { usePageThemeContext } from '~context/PageThemeContext/PageThemeContext.ts';
+
+import { FeaturebaseContext } from './FeaturebaseContext.ts';
+
+const FEATUREBASE_SCRIPT_ID = 'featurebase-sdk';
+
+const FeaturebaseContextProvider: FC<PropsWithChildren> = ({ children }) => {
+  const { user } = useAppContext();
+  const { isDarkMode } = usePageThemeContext();
+
+  const [isFeaturebaseBooted, setIsFeaturebaseBooted] = useState(false);
+
+  const userEmail = useMemo(() => user?.profile?.email, [user]);
+  const displayName = useMemo(() => user?.profile?.displayName, [user]);
+
+  useEffect(() => {
+    if (!document.getElementById(FEATUREBASE_SCRIPT_ID)) {
+      const script = document.createElement('script');
+      script.src = 'https://do.featurebase.app/js/sdk.js';
+      script.id = FEATUREBASE_SCRIPT_ID;
+
+      script.onload = () => {
+        if (typeof window.Featurebase !== 'function') {
+          window.Featurebase = function Featurebase() {
+            // eslint-disable-next-line prefer-rest-params
+            (window.Featurebase.q = window.Featurebase.q || []).push(arguments);
+          };
+        }
+        setIsFeaturebaseBooted(true);
```

Comment:
Would it make sense to extract this into a separate function as we use the same piece of code also below?